### ISSUE TITLE
Consider not using opacity for any text content

### ISSUE THREAD

**Issue by babakfp** [vuejs/vitepress#3313]
### Describe the bug

![Screenshot 2023-12-09 064902](https://github.com/vuejs/vitepress/assets/44144724/02ef7255-3316-41fd-990a-30d049161dfb)

Open the image in a new page and zoom in, you will see the transparency causes the overlaying lines of the letters to be visible and this makes things to look ugly.on

### Reproduction

1. You can use this font to test things out: https://fonts.google.com/specimen/Vazirmatn
2. Change the font-family to this ^^^ font.
3. Change to dark mode.
4. Use this text: `اعتبارسنجی اسکیما تایپ‌اسکریپت-اول`
5. Zoom and you can see it too.

### Expected behavior

So the overlaying paths/lines are not visible.

### System Info

```Text
.
```

### Additional context

I think this doesn't happen with all Farsi/Persian fonts, but it's something that happens very often.

The only solution is to not use opacity with text content.

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by akirarika** [comment]
As a designer, I understand the reason for adding transparency to fonts on a black background. It helps to reduce the harshness and make the text less glaring.

Although it may sound discriminatory to say so, the majority of websites worldwide use English as their language. Therefore, I believe that keeping transparency as the default option is a correct choice, and users of other languages can easily override it by writing CSS.
